CMakeLists.txt

```diff
@@ -230,6 +226,29 @@ else ()
 	find_library(M_LIBRARY m)
 endif ()
 
+if (NOT ENABLE_EXTERNAL_BLOCKS_RUNTIME)
+	set(libBlocksRuntime_COMPATIBILITY_HDRS
+		Block.h
+		Block_private.h
+		)
+	list(APPEND libobjc_OBJC_SRCS blocks_runtime.m)
+	list(APPEND libobjc_HDRS objc/blocks_private.h)
+	list(APPEND libobjc_HDRS objc/blocks_runtime.h)
+	add_definitions(-DEMBEDDED_BLOCKS_RUNTIME)
+else ()
+	find_library(BLOCKS_RUNTIME_LIBRARY BlocksRuntime)
+	if (BLOCKS_RUNTIME_LIBRARY)
+		set(CMAKE_REQUIRED_LIBRARIES ${BLOCKS_RUNTIME_LIBRARY})
+		check_symbol_exists(_Block_use_RR2 "Block_private.h" HAVE_BLOCK_USE_RR2)
+		if (HAVE_BLOCK_USE_RR2)
```

Comment:
Ah, yes absolutely. If forgot to make it fail hard when I  remembered how, uhm, _quirky_ the `objc_constructInstance`/`objc_destructInstance` API is that the legacy hook function would  need.

Reply:
At some point, I am pondering doing the thing Apple does of embedding the refcount and weak flags in the isa pointer. I believe GNUstep should now all be clean to that.

It probably doesn’t make sense to do on 32-bit platforms so it would be 64-bit only, but would give us a nice memory saving there and would make it easier to support those APIs, at least on 64-bit platforms.